### PR TITLE
resource_control: support config background tasks resource limit

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18642
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -2426,6 +2426,16 @@ Configuration items related to resource control of the TiKV storage layer.
 + Enabling this configuration item only works when [`tidb_enable_resource_control](/system-variables.md#tidb_enable_resource_control-new-in-v660) is enabled on TiDB. When this configuration item is enabled, TiKV will use the priority queue to schedule the queued read/write requests from foreground users. The scheduling priority of a request is inversely related to the amount of resources already consumed by the resource group that receives this request, and positively related to the quota of the corresponding resource group.
 + Default value: `true`, which means scheduling based on the RU of the resource group is enabled.
 
+### `priority-ctl-strategy` <span class="version-mark">从 v8.4.0 版本开始引入</span>
```
Comment (hfxsd) [inline tikv-configuration-file.md:2429]:
```suggestion
### `priority-ctl-strategy` <span class="version-mark">New in v8.4.0</span>
```

File: dynamic-config.md [hunk 2]
```diff
@@ -169,6 +169,7 @@ The following TiKV configuration items can be modified dynamically:
 | `readpool.unified.max-thread-count` | The maximum number of threads in the thread pool that uniformly processes read requests, which is the size of the UnifyReadPool thread pool |
 | `readpool.unified.max-tasks-per-worker` | The maximum number of tasks allowed for a single thread in the unified read pool. `Server Is Busy` error is returned when the value is exceeded. |
 | `readpool.unified.auto-adjust-pool-size` | Determines whether to automatically adjust the UnifyReadPool thread pool size |
+| `resource-control.priority-ctl-strategy` | Config the control strategy of low-priority tasks. |
```
Comment (hfxsd) [inline dynamic-config.md:172]:
```suggestion
| `resource-control.priority-ctl-strategy` | Configures the control strategy of low-priority tasks. |
```

File: sql-statements/sql-statement-alter-resource-group.md [hunk 3]
```diff
@@ -82,7 +83,7 @@ TiDB supports the following `DirectResourceGroupOption`, where [Request Unit (RU
...
-| `BACKGROUND`  | Configure the background tasks. For more details, see [Manage background tasks](/tidb-resource-control.md#manage-background-tasks). | `BACKGROUND=(TASK_TYPES="br,stats")` indicates that the backup and restore and statistics collection related tasks are scheduled as background tasks. |
+| `BACKGROUND`  | Configure the background tasks. For more details, see [Manage background tasks](/tidb-resource-control.md#manage-background-tasks). | `BACKGROUND=(TASK_TYPES="br,stats", UTILIZATION_LIMIT=30)` indicates that the backup and restore and statistics collection related tasks are scheduled as background tasks and background tasks can consume 30% of the resources of TiKV at most. |
```
Comment (hfxsd) [inline sql-statements/sql-statement-alter-resource-group.md:86]:
```suggestion
| `BACKGROUND`  | Configure the background tasks. For more details, see [Manage background tasks](/tidb-resource-control.md#manage-background-tasks). | `BACKGROUND=(TASK_TYPES="br,stats", UTILIZATION_LIMIT=30)` indicates that the backup and restore and statistics collection related tasks are scheduled as background tasks, and background tasks can consume 30% of the resources of TiKV at most. |
```

File: tidb-resource-control.md [hunk 4]
```diff
@@ -412,7 +410,8 @@ Starting from v7.4.0, the TiDB resource control feature supports managing backgr
 
 #### `BACKGROUND` parameters
 
-`TASK_TYPES`: specifies the task types that need to be managed as background tasks. Use commas (`,`) to separate multiple task types.
+- `TASK_TYPES`: specifies the task types that need to be managed as background tasks. Use commas (`,`) to separate multiple task types.
+- `UTILIZATION_LIMIT`: specifies the maximum resource percentage(0-100) of each TiKV instance that background tasks can consume. By default, the resource granted for background tasks are calculated based on the total resource quota and current foreground workload dynamically, if `UTILIZATION_LIMIT` is configured, the resource granted for background tasks will not exceed this limit.
```
Comment (hfxsd) [inline tidb-resource-control.md:414]:
```suggestion
- `UTILIZATION_LIMIT`: specifies the maximum resource percentage (0-100) of each TiKV instance that background tasks can consume. By default, the resource granted for background tasks are calculated based on the total resource quota and the current foreground workload dynamically. If `UTILIZATION_LIMIT` is configured, the resource granted for background tasks will not exceed this limit.
```

File: tidb-resource-control.md [hunk 5]
```diff
@@ -444,10 +443,10 @@ By default, the task types that are marked as background tasks are `""`, and the
 
 #### Examples
 
-1. Modify the `default` resource group and mark `br` and `ddl` as background tasks.
+1. Modify the `default` resource group and mark `br` and `ddl` as background tasks and set background tasks resource limit to 30%.
```
Comment (hfxsd) [inline tidb-resource-control.md:446]:
```suggestion
1. Modify the `default` resource group and mark `br` and `ddl` as background tasks, and set background tasks resource limit to 30%.
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -2426,6 +2426,16 @@ Configuration items related to resource control of the TiKV storage layer.
...
+
+Configure the control strategy for low priority tasks. TiKV ensures the priority execution of higher priority tasks by applying dynamic quota limit to low-priority tasks. This control strategy is used to calculates the quota for low priority tasks.
+
++ Value Options:
```
Comment (hfxsd) [inline tikv-configuration-file.md:2433]:
```suggestion
+ Value options:
```

Comment (glorv) [reply tikv-configuration-file.md]: I saw all other places are using `Value Options`

**Review by songrijie — APPROVED** [review]

File: dynamic-config.md [hunk 7]
```diff
@@ -169,6 +169,7 @@ The following TiKV configuration items can be modified dynamically:
 | `readpool.unified.max-thread-count` | The maximum number of threads in the thread pool that uniformly processes read requests, which is the size of the UnifyReadPool thread pool |
 | `readpool.unified.max-tasks-per-worker` | The maximum number of tasks allowed for a single thread in the unified read pool. `Server Is Busy` error is returned when the value is exceeded. |
 | `readpool.unified.auto-adjust-pool-size` | Determines whether to automatically adjust the UnifyReadPool thread pool size |
+| `resource-control.priority-ctl-strategy` | Configures the control strategy of low-priority tasks. |
```
Comment (qiancai) [inline dynamic-config.md:172]:
```suggestion
| `resource-control.priority-ctl-strategy` | Configures the flow control strategy of low-priority tasks. |
```

File: tidb-resource-control.md [hunk 8]
```diff
@@ -412,7 +412,8 @@ Starting from v7.4.0, the TiDB resource control feature supports managing backgr
 
 #### `BACKGROUND` parameters
 
-`TASK_TYPES`: specifies the task types that need to be managed as background tasks. Use commas (`,`) to separate multiple task types.
+- `TASK_TYPES`: specifies the task types that need to be managed as background tasks. Use commas (`,`) to separate multiple task types.
+- `UTILIZATION_LIMIT`: specifies the maximum resource percentage (0-100) of each TiKV instance that background tasks can consume. By default, the resource granted for background tasks are calculated based on the total resource quota and the current foreground workload dynamically. If `UTILIZATION_LIMIT` is configured, the resource granted for background tasks will not exceed this limit.
```
Comment (qiancai) [inline tidb-resource-control.md:416]:
```suggestion
- `UTILIZATION_LIMIT`: limits the maximum percentage  (0-100) of resources that background tasks can consume on each TiKV node. By default, TiKV calculates the available resources for background tasks based on the total resources of the node and the resources currently occupied by the foreground tasks. If `UTILIZATION_LIMIT` is configured, the resource allocated to background tasks will not exceed this limit.
```

File: tidb-resource-control.md [hunk 9]
```diff
@@ -444,10 +445,10 @@ By default, the task types that are marked as background tasks are `""`, and the
 
 #### Examples
 
-1. Modify the `default` resource group and mark `br` and `ddl` as background tasks.
+1. Modify the `default` resource group and mark `br` and `ddl` as background tasks, and set background tasks resource limit to 30%.
```
Comment (qiancai) [inline tidb-resource-control.md:448]:
```suggestion
1. Modify the `default` resource group, mark `br` and `ddl` as background tasks, and set the background tasks resource limit to 30%.
```

File: tikv-configuration-file.md [hunk 10]
```diff
@@ -2426,6 +2426,16 @@ Configuration items related to resource control of the TiKV storage layer.
 + Enabling this configuration item only works when [`tidb_enable_resource_control](/system-variables.md#tidb_enable_resource_control-new-in-v660) is enabled on TiDB. When this configuration item is enabled, TiKV will use the priority queue to schedule the queued read/write requests from foreground users. The scheduling priority of a request is inversely related to the amount of resources already consumed by the resource group that receives this request, and positively related to the quota of the corresponding resource group.
 + Default value: `true`, which means scheduling based on the RU of the resource group is enabled.
 
+### `priority-ctl-strategy` <span class="version-mark">New in v8.4.0</span>
+
+Configure the control strategy for low priority tasks. TiKV ensures the priority execution of higher priority tasks by applying dynamic quota limit to low-priority tasks. This control strategy is used to calculates the quota for low priority tasks.
```
Comment (qiancai) [inline tikv-configuration-file.md:2431]:
```suggestion
Specifies the flow control strategy for low-priority tasks.TiKV ensures that higher priority tasks are prioritized for execution by applying flow control to low-priority tasks.
```

File: tikv-configuration-file.md [hunk 11]
```diff
@@ -2426,6 +2426,16 @@ Configuration items related to resource control of the TiKV storage layer.
...
+
++ Value options:
+    + `aggressive`: using this strategy, the flow control policy will prioritize the performance of high-priority tasks, ensuring that the throughput and latency of high-priority tasks are basically unaffected, but low-priority tasks will run slower.
+    + `moderate`: using this strategy, TiKV will impose a balanced flow control limit on low-priority tasks, ensuring that low-priority tasks can use more system available resources while causing little impact on high-priority tasks.
+    + `conservative`: using this strategy, the flow control policy will prioritize ensuring that system resources are fully utilized, and low-priority tasks will try to use available system resources as much as possible, resulting in a greater impact on the performance of high-priority tasks.
```
Comment (qiancai) [inline tikv-configuration-file.md:2436]:
```suggestion
    + `aggressive`: this policy prioritizes the performance of high-priority tasks, ensuring that the throughput and latency of high-priority tasks are largely unaffected, but low-priority tasks will run slower.
    + `moderate`: this policy imposes a balanced flow control on low-priority tasks and has a lower impact on high-priority tasks.
    + `conservative`: this policy prioritizes ensuring that system resources are fully utilized, allowing low-priority tasks to fully utilize system available resources as needed, and therefore has a greater performance impact on high-priority tasks.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: tidb-resource-control.md [hunk 12]
```diff
@@ -412,7 +412,8 @@ Starting from v7.4.0, the TiDB resource control feature supports managing backgr
 
 #### `BACKGROUND` parameters
 
-`TASK_TYPES`: specifies the task types that need to be managed as background tasks. Use commas (`,`) to separate multiple task types.
+- `TASK_TYPES`: specifies the task types that need to be managed as background tasks. Use commas (`,`) to separate multiple task types.
+- `UTILIZATION_LIMIT`: limits the maximum percentage  (0-100) of resources that background tasks can consume on each TiKV node. By default, TiKV calculates the available resources for background tasks based on the total resources of the node and the resources currently occupied by the foreground tasks. If `UTILIZATION_LIMIT` is configured, the resource allocated to background tasks will not exceed this limit.
```
Comment (glorv) [inline tidb-resource-control.md:416]:
```suggestion
- `UTILIZATION_LIMIT`: limits the maximum percentage (0-100) of resources that background tasks can consume on each TiKV node. By default, TiKV calculates the available resources for background tasks based on the total resources of the node and the resources currently occupied by the foreground tasks. If `UTILIZATION_LIMIT` is configured, the resource allocated to background tasks will not exceed this limit.
```

**Comment by glorv** [comment]
@nolouch PTAL

**Review by hfxsd — APPROVED** [review]

**Pull request merged**